dali/test/python/test_dali_tf_dataset_mnist.py

```diff
@@ -224,15 +225,24 @@ def clear_checkpoints():
 
 def _test_estimators_single_device(model, device='cpu', device_id=0, fail_on_device_mismatch=True):
     def train_fn():
-        with tf.device('/{0}:{1}'.format(device, device_id)):
-            return _get_train_dataset(device, device_id, fail_on_device_mismatch=fail_on_device_mismatch).map(
-                lambda features, labels: ({'images': features}, labels))
+        # with tf.device('/{0}:{1}'.format(device, device_id)):
+        #     return _get_train_dataset(device, device_id, fail_on_device_mismatch=fail_on_device_mismatch).map(
+        #         lambda features, labels: ({'images': features}, labels))
+        ds = _get_train_dataset(device, device_id, fail_on_device_mismatch=fail_on_device_mismatch)
+        return ds.apply(tf.data.experimental.copy_to_device(
```

Comment:
I don't think it is the way. It kills the idea to process data on the GPU if you will copy it to the CPU after all.

Reply:
it's not that "it is" or "it is not" the way. What you guys want to do is just not possible.
the `map()` operation does not exist on GPU. I gave to @awolant the one that is compatible for testing. But it's an undocumented API. I never tried it myself.
This unittest is wrong 100%. Your code is not "legal". This PR fixes that. Now I agree it's kinda non-sense. But saying that doesn't help fixing the unittest ...

Reply:
> Now I agree it's kinda non-sense. 

Then it is better to remove the wrong test that providing a wrong fix to a wrong test.

Reply:
The fix is not "wrong". It fix the issue. However it might not test what you guys want to test ;) 
However, except if @awolant finds how to use the MapOnGPU OP, this is the "right" to execute `map()` after a DALI dataset ;)

Reply:
To keep it short there was a number of issues here:
- only train_fn was using a device scope, not test_fn
- the device scope was automatically applied to MAP operation which doesn't exists on GPU, you can't do this
- Since DALI is on GPU device. Between the DALIDataset and the map() OP you need to re-copy the dataset on CPU device

And here is the MAP on GPU OP: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/ops/prefetching_ops.py#L228

Reply:
Alternatively, can we somehow live without using the `map`? Just make the model accept DALI outputs?

Reply:
You could. But there's still an issue with TF iterator.
`iterator.get_next()` still triggers a memcpy from GPU to CPU and then from CPU to GPU...